ext/js/pages/settings/anki-deck-generator-controller.js

```diff
@@ -0,0 +1,588 @@
+        /** @type {?import('./modal.js').Modal} */
+        this._sendToAnkiConfirmModal = null;
+        /** @type {?import('./modal.js').Modal} */
+        this._exportConfirmModal = null;
+        /** @type {boolean} */
+        this._cancel = false;
+        /** @type {boolean} */
+        this._inProgress = false;
+        /** @type {AnkiNoteBuilder} */
+        this._ankiNoteBuilder = new AnkiNoteBuilder(settingsController.application.api, new TemplateRendererProxy());
+    }
+
+    /** */
+    async prepare() {
+        this._defaultFieldTemplates = await this._settingsController.application.api.getDefaultAnkiFieldTemplates();
+
+        /** @type {HTMLButtonElement} */
+        const testRenderButton = querySelectorNotNull(document, '#generate-anki-notes-test-render-button');
+        /** @type {HTMLButtonElement} */
+        const sendToAnkiButton = querySelectorNotNull(document, '#generate-anki-notes-send-to-anki-button');
+        /** @type {HTMLButtonElement} */
+        const sendToAnkiCancelButton = querySelectorNotNull(document, '#generate-anki-notes-send-to-anki-cancel-button');
+        /** @type {HTMLButtonElement} */
+        const exportButton = querySelectorNotNull(document, '#generate-anki-notes-export-button');
+        /** @type {HTMLButtonElement} */
+        const exportCancelButton = querySelectorNotNull(document, '#generate-anki-notes-export-cancel-button');
+        /** @type {HTMLButtonElement} */
+        const generateButton = querySelectorNotNull(document, '#generate-anki-notes-export-button');
+
+        this._sendToAnkiConfirmModal = this._modalController.getModal('generate-anki-notes-send-to-anki');
+        this._exportConfirmModal = this._modalController.getModal('generate-anki-notes-export');
+
+        testRenderButton.addEventListener('click', this._onRender.bind(this), false);
+        sendToAnkiButton.addEventListener('click', this._onSendToAnki.bind(this), false);
+        this._sendToAnkiButtonConfirmButton.addEventListener('click', this._onSendToAnkiConfirm.bind(this), false);
+        sendToAnkiCancelButton.addEventListener('click', (() => { this._cancel = true; }).bind(this), false);
+        exportButton.addEventListener('click', this._onExport.bind(this), false);
+        this._exportButtonConfirmButton.addEventListener('click', this._onExportConfirm.bind(this), false);
+        exportCancelButton.addEventListener('click', (() => { this._cancel = true; }).bind(this), false);
+        generateButton.addEventListener('click', this._onExport.bind(this), false);
+
+        void this._updateExampleText();
+        this._mainSettingsEntry.addEventListener('click', this._updateExampleText.bind(this), false);
+
+        void this._updateActiveModel();
+        this._mainSettingsEntry.addEventListener('click', this._updateActiveModel.bind(this), false);
+    }
+
+    // Private
+
+    /** */
+    async _updateActiveModel() {
```

Comment:
What happens when Anki is not open?

Reply:
Exporting to file still works perfectly fine, sending to anki throws an anki connection error visible to the user https://github.com/themoeway/yomitan/pull/895#issuecomment-2104779223.